browser_use/agent/service.py

```diff
@@ -531,12 +545,24 @@ async def get_next_action(self, input_messages: list[BaseMessage]) -> AgentOutpu
 
 		elif self.tool_calling_method is None:
 			structured_llm = self.llm.with_structured_output(self.AgentOutput, include_raw=True)
-			response: dict[str, Any] = await structured_llm.ainvoke(input_messages)  # type: ignore
-			parsed: AgentOutput | None = response['parsed']
+			try:
+				response: dict[str, Any] = await structured_llm.ainvoke(input_messages)  # type: ignore
+				parsed: AgentOutput | None = response['parsed']
+
+			except Exception as e:
+				logger.error(f'Failed to invoke model: {str(e)}')
+				raise LLMException(401, 'Failed to invoke model')
```

Comment:
```suggestion
				raise LLMException(401, 'Failed to invoke model') from e
```
always preserve the original error in the traceback with `from e`